webdriver-spec.html

```diff
@@ -5834,36 +5841,25 @@
  <li><p>Let <var>body</var> and <var>arguments</var>
   be <var>script arguments</var>’ data.
 
- <li><p class=issue>This next step might not quite set up all the right machinery.
+  <li><p>Let <var>promise</var> be a new Promise.
 
-  <p>Let <var>webdriver callback result</var> be
-   a flag which can have three values:
-   <code>unset</code>, <code>expired</code>, or <code>set</code>,
-   with the <code>set</code> value having associated data.
-   Initially it is in the <code>unset</code> state.
-
- <li><p>Let <var>callback</var> be a <a>function</a>
-  whose <a>[[\Call]]</a> <a>internal method</a> runs
-  the <a>execute async script callback</a> algorithm
-  initialized with argument <var>webdriver callback result</var>.
-
- <li><p>Append <var>callback</var> to <var>arguments</var>.
-
- <li><p>Let <var>result</var> be the result
-  of <a>trying</a> to call <a>execute a function body</a>
-  with arguments <var>body</var> and <var>arguments</var>.
+  <li><p>Run the following substeps <a>in parallel</a>:
+  <ol>
+    <li><p>Append resolve to <var>script arguments</var>.
 
- <li><p>Wait for <var>webdriver callback result</var> to enter the <code>set</code> state,
-  or for <a>session script timeout</a> milliseconds to expire,
-  whichever happens sooner.
+    <li><p>Let <var>result</var> be the result of calling <a>execute a function body</a>,
+     with arguments <var>body</var> and <var>arguments</var>.
 
- <li><p>If the previous step completed due the <a>session script timeout</a> being reached,
-  set <var>webdriver callback result</var> to <code>expired</code>
-  and return <a>error</a> with <a>error code</a> <a>script timeout</a>.
+    <li><p>If <var>promise</var> is still pending and
+      <a>session script timeout</a> milliseconds is reached,
+     reject <var>promise</var> with <a>error</a> with
+     <a>error code</a> <a>script timeout</a>.
 
- <li><p>Otherwise, let <var>result</var> be <var>webdriver callback result</var>’s data.
+    <li><p>If <var>result</var> is an <a>error</a>,
```

Comment:
Same comment as synchronous script: sounds like you're requiring an instanceof check instead of a try-catch